node/src/utils.rs

```diff
@@ -7,6 +7,7 @@ pub(crate) mod ds;
 mod external;
 pub(crate) mod fmt_limit;
 pub(crate) mod opt_display;
+pub(crate) mod pid_file;
```

Comment:
Just a question, because I'm not fan o pid files :)
Would it be possible to keep a state in the local storage?

Reply:
It would be possible. However, we went with the pid file approach because it is more straightforward for operators to interact with this switch. If we went with the storage approach, we would have to provide dedicated tooling in order to interact with this switch, which has merit but wasn't something we were willing to do right now.

Reply:
After further discussion, we decided to use a regular file instead of a pid file as it meets our requirements and is simpler to implement and use.